packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -1,17 +1,46 @@
 import type { LoadedStrapi } from '@strapi/types';
 import { omit, pick } from 'lodash/fp';
+
+import { scheduleJob } from 'node-schedule';
 import { HISTORY_VERSION_UID } from '../constants';
 
 import type { HistoryVersions } from '../../../../shared/contracts';
 
+const DEFAULT_RETENTION_DAYS = 90;
+
 const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
+  const state: {
+    deleteExpiredJob: ReturnType<typeof scheduleJob> | null;
+  } = {
+    deleteExpiredJob: null,
+  };
+
   /**
    * Use the query engine API, not the document service,
    * since we'll refactor history version to be just a model instead of a content type.
    * TODO: remove this comment once the refactor is done.
    */
   const query = strapi.db.query(HISTORY_VERSION_UID);
 
+  const getRetentionDays = (strapi: LoadedStrapi) => {
+    const licenseRetentionDays =
+      strapi.ee.features.get('cms-content-history')?.options.retentionDays;
+    const userRetentionDays = strapi.config.get('admin.history.retentionDays');
+
+    // For enterprise plans, use 90 days by default, but allow users to override it
+    if (licenseRetentionDays == null) {
+      return userRetentionDays ?? DEFAULT_RETENTION_DAYS;
+    }
```

Comment:
Actually after discussing with Mikkel we chose to make it a very large number for EE, not null. Similar to content releases I believe